bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,508 @@
+            modelImporter.import(dataToImport)
+        }
+
+        val expectedErrorMessage = "Unexpected concept change for node with ID `child1` from `uid1` to `uid2`."
+        assertEquals(expectedErrorMessage, actualException.message)
+    }
+
+    @Test
+    fun `children without resolvable role are ordered`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "child1",
+                "role": "unknownRole"
+              },
+              {
+                "id": "child2",
+                "role": "unknownRole"
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "child2",
+                "role": "unknownRole"
+              },
+              {
+                "id": "child1",
+                "role": "unknownRole"
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.toList()
+        // child1 and child 2 there role cannot be resolved
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
+    }
+
+    @Test
+    fun `children without role are ordered`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "child1"
+              },
+              {
+                "id": "child2"
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "child2"
+              },
+              {
+                "id": "child1"
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.toList()
+        // child1 and child 2 are reordered because they have no role.
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
+    }
+
+    @Test
+    fun `unordered children are synced recursively`() {
```

Comment:
Do we also need this test case for ordered children?

Reply:
For the ordered children the test already existed in `ModelImporterTest.kt`.
As I now use the same code path as for ordered children, I removed this seperate test.